models/etablissement.ts

```diff
@@ -24,31 +33,107 @@ const getEtablissementFromSlug = async (
   const siret = verifySiret(slug);
 
   const isBot = options?.isBot || false;
-  const shouldNotUseInsee = process.env.INSEE_ENABLED === 'disabled';
 
-  const etablissement =
-    shouldNotUseInsee || isBot
-      ? await getEtablissementRechercheEntreprise(siret)
-      : await getEtablissement(siret);
+  const etablissement = fetchFromClients(siret, isBot);
 
   return etablissement;
 };
 
 /**
  * Return an Etablissement for a given siret
  */
-const getEtablissement = async (siret: Siret): Promise<IEtablissement> => {
+const fetchFromClients = async (
+  siret: Siret,
+  isBot = false
+): Promise<IEtablissement> => {
+  // no cache for bot as they scrap so they tend not to call the same siren twice
+  const useCache = !isBot;
+
+  const etablissementRechercheEntreprise =
+    await fetchEtablissementFromRechercheEntreprise(siret, useCache);
+
+  const useInsee = shouldUseInsee(etablissementRechercheEntreprise, isBot);
+
+  if (!useInsee) {
+    if (isAPINotResponding(etablissementRechercheEntreprise)) {
+      throw new HttpServerError('Recherche failed, return 500');
+    }
+    return etablissementRechercheEntreprise;
+  }
+
+  const etablissementInsee = await fetchEtablissmentFromInsee(siret, {
+    useFallback: false,
+    useCache,
+  });
+
+  if (isAPINotResponding(etablissementInsee)) {
+    if (isAPINotResponding(etablissementRechercheEntreprise)) {
+      const etablissmentInseeFallbacked = await fetchEtablissmentFromInsee(
+        siret,
+        {
+          useFallback: true,
+          useCache,
+        }
+      );
+      if (isAPINotResponding(etablissmentInseeFallbacked)) {
+        throw new HttpServerError('Sirene Insee fallback failed, return 500');
+      }
+      return etablissmentInseeFallbacked;
+    } else {
+      return etablissementRechercheEntreprise;
+    }
+  } else {
+    if (isAPINotResponding(etablissementRechercheEntreprise)) {
+      return etablissementInsee;
+    } else {
+      return {
+        ...etablissementInsee,
+        complements: {
+          ...etablissementInsee?.complements,
+          ...etablissementRechercheEntreprise.complements,
+        },
+      };
+    }
+  }
+};
+
+const shouldUseInsee = (
+  etablissementRechercheEntreprise: IEtablissement | IAPINotRespondingError,
+  isBot: boolean
+) => {
+  const isInseeEnabled = process.env.INSEE_ENABLED !== 'disabled';
+
+  if (!isInseeEnabled) {
+    return false;
+  }
+
+  const rechercheEntrepriseFailed = isAPINotResponding(
+    etablissementRechercheEntreprise
+  );
+
+  if (rechercheEntrepriseFailed) {
+    return true;
+  } else {
+    if (isBot) {
+      return false;
+    }
+
+    // we always call insee for etablissement as we dont know if they belong to an EI
+    return true;
```

Comment:
We don't know, but we could know, as the SearchAPI does return if it's an EI or not. We could : 
- Adapt the `clientEtablissementRechercheEntreprise` client to return this information
- Adapt the model to call for `uniteLegale` (with the cache logic, it should not do more than one call).

Reply:
Either way, we could add a `isEi` callback to the orchestrator.